document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -82,6 +83,38 @@ public boolean upsert(Key key, Document document) throws IOException {
     }
   }
 
+  @Override
+  public boolean upsert(Key key, Document document, Filter condition, @Nullable Boolean isUpsert)
+      throws IOException {
+    StringBuilder upsertQueryBuilder =
+        new StringBuilder(getUpsertSQL(isUpsert != null ? isUpsert : true));
+
+    String jsonString = prepareUpsertDocument(key, document);
+    Params.Builder paramsBuilder = Params.newBuilder();
+    paramsBuilder.addObjectParam(key.toString());
+    paramsBuilder.addObjectParam(jsonString);
+    paramsBuilder.addObjectParam(jsonString);
+
+    if (condition != null && isUpsert == true) {
+      String conditionQuery = parseFilter(condition, paramsBuilder, "d");
```

Comment:
Can you please add a comment about the argument `d` here?

Reply:
No longer needed.